src/logic/ClientController.ts

```diff
@@ -567,10 +575,16 @@ export default class ClientController {
 						console.error(url + ' ' + getErrorMessage(e));
 					}
 				},
-				index * 1000
+				index * timeoutValue
 			);
 		});
 
+		window.setTimeout(() => {
+			if(failedLoadsToRetry.length > 0){
+				this.onRoundContent(failedLoadsToRetry);
```

Comment:
Provide retry count limit. Current implementation could possibly run indefinitely

Reply:
@VladimirKhil 
Sure.
Aside of this PR I was looking into storage server implementation, and see that you're using ASP .NET static file handler, why don't just replace it with pure nginx(since you already have it) and add proper cachig? Won't it reduce number of errors?
Also, as an option can be usage of some CDN(even free plans should help)

Reply:
Nginx is already used in prod (see https://github.com/VladimirKhil/SIContentService - you could suggest changes to enable caching there; it would be great). ASP .NET static file handler is used only for debugging.
What free CDN could you suggest? Game traffic is around 2 TB per day.